doc/usage/usage.rst

```diff
@@ -1,15 +1,238 @@
+
+    ./pipeline run_event --settings pipeline.ini "event specification"
+    ./run_event --settings pipeline.ini "event specification"
+
+Plugins
+^^^^^^^
+
+Pipeline plugins are python packages that extend the functionality of the Permian
+pipeline. Plugins can be located outside of the libpermian directory.
+
+Some plugins may be disabled by default (empty ``DISABLED`` file in the plugin's directory).
+Disabled plugins can be enabled with environment variable ``PIPELINEPLUGINS_ENABLE``, and plugins
+enabled by default can be disabled with variable ``PIPELINEPLUGINS_DISABLE``. Plugin names
+are separated with comma. ::
+
+    PIPELINEPLUGINS_ENABLE=test PIPELINEPLUGINS_DISABLE=beaker,beaker_tag ./pipeline ...
+
+**Additional plugins**
+
+Permian can load aditional plugins from locations defined in environment variable
+``PIPELINEPLUGINS_PATH``, paths are separated by colon::
+
+    PIPELINEPLUGINS_PATH=../my_plugins/plugins/:/home/user/secret_plugins/ ./pipeline ...
+
+If you are running pipeline in container, you have to specify paths inside the cotainer
+and they have to be accessible there. Default ``./in_container`` script doesn't help with this.
+
+Running only subsets of tests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Just use run_subset command and specify desired limiting factors before the
+original command (see ``./run_subset --help``). The resulting command line will look like::
+
+    ./run_subset --configuration architecture:s390x -s pipeline.ini compose ...
+
+To run a subset of tests verifying a specific requirement and filtering the list of test cases::
+
+    ./run_subset --testcase-query '"installer" in tc.tags and "disabled" not in tc.tags and tc.priority < 9 and tc.execution.type != "manual" and "RHEL-8 Installation: Driverdisk" in tc.verifiesRequirement|map(attribute="name")' -s pipeline.ini compose ...
+
+Dry run
+^^^^^^^
+Partial dry run, execute tests, but don't report the results::
+
+    -o reportSenders.dry_run=True
+
+Full dry run, don't execute tests and don't report results::
+
+    -o workflows.dry_run=True -o reportSenders.dry_run=True
+
+
+Local test library
+^^^^^^^^^^^^^^^^^^
+Running pipeline on local testplans library (e.g. on not committed or not merged changes)::
+
+    -o library.directPath=./path/to/library
+
+If you are running pipeline in container the library must be a subdirectory of the
+permian (cwd) directory so that it is reachable inside the container.
+
+Events
+------
+Events are describing the cause (or impulse) for the pipeline to start and
+based on the event all the actions taken by the pipeline are taken.
+
+One of the main functions of the events is to provide name of branch which
+contains testplans that should be executed for such event instance.
+
+Events may also require or provide additional information related to the event
+instance where such information can be taken from external sources and is stored
+in event structures.
```

Comment:
I do not understand the sentence. Can you update it and/or provide an example?

Reply:
I've added an example right bellow the sentence.